CCPAPI/CMakeLists.txt

```diff
@@ -0,0 +1,174 @@
+##########################################################################
```

Comment:
Since we are making an effort to update cmake and to use the modern style, these cmake files should be updated to take at least [these dos and don'ts](https://cliutils.gitlab.io/modern-cmake/chapters/intro/dodonot.html) into account.

At a quick glance:
- use lower case for cmake functions
- use targets for include_directories (target_include_directories)
- don't use GLOB

Reply:
Done!

Reply:
Thanks - I'll have to look at them more closely when you reorganize the files. I only took a quick look at it. I'll try not to hold it up if things can be fixed later in a separate PR.

I would suggest taking a look at the rest of that [An Introduction to Modern CMake](https://cliutils.gitlab.io/modern-cmake/) document. It's relatively short and gives solid recommendations and examples of what cmake should look like.

Before I found it, cmake was just a blob of unstructured stuff that never really made sense to me :-)

Reply:
Thank you for [An Introduction to Modern CMake](https://cliutils.gitlab.io/modern-cmake/), very interesting: I think I will keep it on my bedside table :-)

Reply:
Other good reads if you're "enjoying" CMake 😀:
- [Enhanced source file handling with target_sources()
](https://crascit.com/2016/01/31/enhanced-source-file-handling-with-target_sources/)
- [Generated Sources In CMake Builds](https://crascit.com/2017/04/18/generated-sources-in-cmake-builds/)